packages/fx-core/src/core/middleware/utils/debug/appLocalYmlGenerator.ts

```diff
@@ -1,16 +1,43 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { ProjectSettings } from "@microsoft/teamsfx-api";
+import { AzureSolutionSettings, ProjectSettings } from "@microsoft/teamsfx-api";
 import { BuildArgs } from "../../../../component/driver/interface/buildAndDeployArgs";
 import { InstallToolArgs } from "../../../../component/driver/tools/interfaces/InstallToolArgs";
 import { BaseAppYmlGenerator } from "../appYmlGenerator";
 
 export class AppLocalYmlConfig {
+  registerApp?: {
```

Comment:
What about organize attributes according to project type?
```
{
    tab?: {
        sso?: boolean;
        domain?: string;
        endpoint?: string;
        port?: number;
    },
    bot?: {
          ......
    }
}
```

Reply:
If so, the syntax in yml template will be complex. For example, to check if `aadApp/create` is needed, we should check `tab.sso` and `bot.sso`.